src/api/query-helpers.ts

```diff
@@ -222,3 +223,55 @@ export function validatePrincipal(stxAddress: string) {
     );
   }
 }
+
+export function parseAddressOrTxId(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): { address: string; txId: undefined } | { address: undefined; txId: string } | never {
+  const address = req.query.address;
+  const txId = req.query.txId;
+  if (!address && !txId) {
+    handleBadRequest(res, next, `can not find address or txId the request`);
+  }
+  if (address && txId) {
+    //if mutually exclusive address and txId specified throw
+    handleBadRequest(res, next, `can't handle both 'address' and 'txId' in the same request`);
+  }
+  if (address) {
+    if (typeof address === 'string') return { address, txId: undefined };
+    handleBadRequest(res, next, `invalid 'address'`);
+  }
+  if (typeof txId === 'string') return { address: undefined, txId };
+  handleBadRequest(res, next, `invalid 'txId'`);
+}
+
+export function parseEventTypeFilter(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): DbEventTypeId[] {
+  const typeQuery = req.query.type;
+  let eventTypeFilter: DbEventTypeId[];
+  try {
+    if (Array.isArray(typeQuery)) {
+      eventTypeFilter = parseEventTypeStrings(typeQuery as string[]);
+    } else if (typeof typeQuery === 'string') {
+      eventTypeFilter = parseEventTypeStrings([typeQuery]);
+    } else if (typeQuery) {
+      handleBadRequest(res, next, `invalid 'event type format'`);
+    } else {
+      eventTypeFilter = [
+        DbEventTypeId.SmartContractLog,
+        DbEventTypeId.StxAsset,
+        DbEventTypeId.FungibleTokenAsset,
+        DbEventTypeId.NonFungibleTokenAsset,
+        DbEventTypeId.StxLock,
+      ]; //no filter provided , return all types of events
+    }
+  } catch (error) {
+    handleBadRequest(res, next, `invalid 'event type'`);
```

Comment:
The try/catch around this whole block is meant to handle the error thrown by `parseEventTypeStrings()` right? If so, it should be scoped to that function call such that it doesn't overlap with the other `handleBadRequest` instead of the try body.